doc/reverse_engineering/networking/13-other.md

```diff
@@ -0,0 +1,351 @@
+
+Used for resigning and disconnecting.
+
+### Definition
+
+```ruby
+def Resign
+  int8 :action_identifier
+  int8 :player_number
+  int8 :player_id
+  int32 :disconnect
+end
+```
+
+### Description
+
+*:action_identifier*  
+Always has the value `0x0b`.
+
+*:player_number*  
+The player's number which can be changed in the lobby. Is only different from *:player_id* if playing in coop mode.
+
+*:player_id*  
+The ID of the player.
+
+*:disconnect*  
+Determines whether the player gave up himself (`0x00`) or was disconnected (`0x01`).
+
+### Examples
+
+`0b 02 03 00`
+
+>`0b` &mdash; action_identifier  
+>`02` &mdash; player_number  
+>`03` &mdash; player_id  
+>`00` &mdash; disconnect
+
+## Save game
+
+Used for saving the game.
+
+### Definition
+
+```ruby
+def Save
+  int8 :action_identifier
+  int8 :exit
+  int8 :player_id
+  unknown :data
+end
+```
+
+### Description
+
+*:action_identifier*  
+Always has the value `0x1b`.
+
+*:exit*  
+Represents whether this action is a normal Save (`0x00`) or a Save & Exit (`0x01`).
+
+*:player_id*  
+The ID of the player saving the game.
+
+*:data*  
+A 253 byte long data field which cannot be parsed easily. Contains filename and presumably a checksum.
+
+### Examples
+
+```
+0000   1b 00 01 78 79 7a 2e 6d 73 78 00 7b 78 79 7a 00
+0010   00 00 00 00 7e cd 83 7b 5b f3 83 7b 00 8c ff 3f
+0020   00 00 00 80 03 00 00 00 b0 d7 32 00 03 00 00 00
+0030   80 00 00 00 00 00 00 00 00 00 00 00 03 00 00 00
+0040   b0 d7 32 00 03 00 00 00 00 00 00 80 00 00 00 00
+0050   06 f3 83 7b 88 d7 32 00 09 00 00 00 80 00 00 00
+0060   ff ff ff ff cc d7 32 00 63 51 61 00 00 00 00 00
+0070   63 51 61 00 3c d8 32 00 00 00 00 80 03 00 00 00
+0080   b0 d7 32 00 03 00 00 00 87 51 61 00 02 00 00 00
+0090   d2 ed 66 00 01 00 00 00 00 00 00 00 0c 00 00 00
+00a0   00 00 00 00 01 00 00 00 03 00 00 00 00 00 00 80
+00b0   03 00 00 00 16 00 00 00 f8 d7 32 00 9d 94 61 00
+00c0   3c d8 32 00 00 80 00 00 40 00 00 00 a4 01 00 00
+00d0   54 6d 14 0b 50 65 14 0b 58 6e 14 0b 00 00 00 00
+00e0   00 00 00 00 b0 d7 79 0c 9d 39 61 00 3c d8 32 00
+00f0   d0 ed 66 00 40 00 00 00 08 1b 68 00 b0 39 61 00
+0100   3c d8 32 00 d0 ed 66 00 2c 00 00 00
+```
+
+>`1b` &mdash; action_identifier  
+>`00` &mdash; exit  
+>`01` &mdash; player_id  
+> data
+
+## Diplomacy, Cheats and Gamespeed
+
+Used for diplomacy, cheating and changing the game's speed.
+
+### Definition
+
+```ruby
+def Diplomacy
+  int8 :action_identifier
+  int8 :action_type
+  int8 :source_player_number
+  int8 :zero
+  int8 :option
+  byte24 :zero2
+  float :option2
+  int8 :diplomatic_stance
+  byte24 :zero3
+end
+```
+
+### Description
+
+*:action_identifier*  
+Always has the value `0x67`.
+
+*:action_type*  
+Represents the specific action that was taken (diplomatic, changing speed or cheating).
+
+Hex Value | Action
+----------|-------
+0x00      | Diplomacy
+0x01      | Change Game Speed
+0x04      | Cheat Response (if cheat is activated for all players, e.g. "aegis")
```

Comment:
I think this is _only_ used for "aegis". It's called "quick build" internally. The packets should look like `0x67 0x04 0x01` when quick build is enabled, and `0x67 0x04 0x00` when quick build is disabled. (toggles every time you use "aegis").

Reply:
This agrees with my research.

Reply:
Yes, that's true. Have to update the description of *:source_player_number* to reflect that. This action drove me crazy because of its multi purpose values :D